multimodal/src/autogluon/multimodal/configs/optimization/adamw.yaml

```diff
@@ -20,16 +21,21 @@ optimization:
   top_k_average_method: "greedy_soup"  # We support averaging method described in https://arxiv.org/pdf/2203.05482.pdf.
                                        # Currently support "uniform_soup", "greedy_soup", and "best".
   efficient_finetune:  # Can be 'bit_fit' (only finetune bias), 'norm_fit' (finetune the normalization terms + bias terms), lora (LoRA Adaptations only), lora_bias (LoRA Adaptation + bit_fit), lora_norm (LoRA Adaptation + norm_fit), or null
+  trainable_param_names: # If efficient_finetune is not None, specifies which layers should be trained.
```

Comment:
1.  The layers (`head`, `fusion_mlp`, and `adapter`) depends on our model definitions. It may not suitable for users to customize them.
2. The name `trainable_param_names ` looks over-general. It is used only in the efficient finetuning mode, but its name seems indicating more than this, which can cause confusions.

Therefore, I suggest we may need to remove this flag from config.

Reply:
Actually, if we want to automatically detect such layers, we can check if layer id == 0. Layers with id 0 are usually what we create. You could refer to the `.get_layer_ids()` in each model. This way of detecting trainable layers are more general than matching the names (`head`, `fusion_mlp`, and `adapter`).

Reply:
I see, agreed. Is there a reason that we assume these layers should always be fine-tuned? In T-Few the LM-head is frozen and in some experiments I've run, freezing the head performs much better (also the head itself adds around 70M parameters in T0-3B). Maybe I can add a binary flag to the config, whether id 0 should be fine-tuned?

Reply:
If you check `HFAutoModelForTextPrediction` and `MultimodalFusionMLP` definitions, you can find that these layers (`head`, `fusion_mlp`, and `adapter`) are created by us instead of from the pretrained model. Since these layers are randomly initialized, we need to fine-tune them. 

You can use the layer id ==0 to determine which layers to fine-tune. If you don't want to fine-tune some layers, just make their ids >=1.

Reply:
Ah, makes total sense. Sounds good.